plugins/travis-ci/src/components/Settings/Settings.tsx

```diff
@@ -56,6 +56,22 @@ const Settings = () => {
 
   const [saved, setSaved] = useState(false);
 
+  const handleSaveSettings = useCallback(() => {
+    setSaved(true);
+    saveSettings({ repo, owner, token });
+    hideSettings();
+  }, [repo, owner, token, setSaved, saveSettings, hideSettings]);
+
+  const handleKeyDown = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.which === 13 && token && owner && repo) {
+        e.preventDefault();
+        handleSaveSettings();
+      }
+    },
+    [token, owner, repo, handleSaveSettings],
+  );
+
   return (
     <>
       <Snackbar
```

Comment:
If you put the text fields into a form and then bind to the `onSubmit` event of the form it should handle both button clicks and the user pressing the Enter key.

Like this: https://stackoverflow.com/a/33212911